### DISCUSSION TITLE
Future features for Plugma

### DISCUSSION THREAD

**Discussion by gavinmcfarland** [gavinmcfarland/plugma#17] [Ideas]
This is a place to suggest  ideas for future releases of Plugma.

Please limit one idea per comment.

**Comment by gavinmcfarland** [comment]
Better create command that's dynamic and supports options like, typescript, eslinting, plugin types (e.g. with and without UI)

**Comment by gavinmcfarland** [comment]
Ability to inspect any node on the canvas and pin certain properties

**Comment by gavinmcfarland** [comment]
Support for individual screens for each plugin command

**Comment by gavinmcfarland** [comment]
Support for widgets

**Comment by gavinmcfarland** [comment]
Ability to change theme mode from browser preview

**Comment by gavinmcfarland** [comment]
Customisable toolbar so developers can configure their own options

**Comment by gavinmcfarland** [comment]
Custom console inside plugin that can run the Figma API and echo results

**Comment by gavinmcfarland** [comment]
A banner that lets you know the UI is running on local dev server

**Comment by m1o191** [comment]
First, thanks for your awesome contribution to the figma plugin dev community!
I would like to have the ability to provide some code to be injected in the head tag of the UI, currently, there is no index.html file, but it's important to be able to add code to the head tag, for example, when applying dark theme support based on multiple variables such as the local storage and the user's preferred schema, to avoid FOUC, the code that adds the dark class needs to be in the head.

**Reply by gavinmcfarland** [reply]
Hi, thanks for your feedback! Good news—this is already built into V2. Things are still evolving, but if you'd like to try it out, you can install it with:

```bash
npm install plugma@next
```

The only breaking change I’m aware of is that you'll need to update your manifest to use a wildcard for WebSockets.

You can find the migration guide here:
[Plugma V2 Migration Guide](https://github.com/gavinmcfarland/plugma/tree/refactor/version-2-beta/packages/plugma/migration/v2)

**Comment by m1o191** [comment]
Thanks, this came in time! The testing feature looks promising too.
